lib/modules/manager/tekton/extract.ts

```diff
@@ -0,0 +1,97 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { TektonBundle, TektonResource } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: TektonResource[];
+  try {
+    docs = loadAll(content) as TektonResource[];
+  } catch (err) {
+    logger.debug(
+      { err, fileName },
+      'Failed to parse YAML resource as a Tekton resource'
+    );
+    return null;
+  }
+  for (const doc of docs) {
```

Comment:
```suggestion
  if (!is.array(docs)) {
    return null;
  }

  for (const doc of docs) {
```
i think `loadAll` returns null for an empty string. please validate with a test.

Reply:
Added unit test to verify behavior when file is empty (empty string). `loadAll` does not return null. I believe the function signature also confirms this:
```typescript
export function loadAll(str: string, iterator?: null, opts?: LoadOptions): unknown[];
```

The suggested change would introduce unreachable code thus failing to meet the required 100% unit test coverage.